workspaces/adventure-pack/goodies/typescript/Iterator.prototype.drop/index.test.ts

```diff
@@ -0,0 +1,78 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).drop;
+
+import "./index";
+
+describe("Iterator.prototype.drop", () => {
+  it("drops the specified number of elements from the iterator", () => {
+    const result = Array.from([10, 22, 33, 47, 58].values().drop(3));
+    expect(result).toStrictEqual([47, 58]);
+  });
+
+  it("returns an empty iterator if the specified number is greater than the iterator length", () => {
+    const result = Array.from([10, 22, 33].values().drop(5));
+    expect(result).toStrictEqual([]);
+  });
+
+  it("returns the full iterator if the specified number is 0", () => {
+    const result = Array.from(["cat", "dog", "otter", "bear"].values().drop(0));
+    expect(result).toStrictEqual(["cat", "dog", "otter", "bear"]);
+  });
+
+  it("returns an empty iterator for an empty iterator", () => {
+    const result = Array.from([].values().drop(3));
+    expect(result).toStrictEqual([]);
+  });
+
+  it("drops the specified number of elements from an infinite iterator", () => {
+    function* infiniteNumbers() {
+      while (true) {
+        yield Math.floor(Math.random() * 100);
+      }
+    }
+
+    // Drop the first 2 elements, then take the next 5 to avoid infinte loop
+    const iterator = infiniteNumbers().drop(2).take(5);
+    const result = [];
+
+    for (const num of iterator) {
+      result.push(num);
+    }
+
+    expect(result.length).toBe(5);
+    expect(result.every((num) => typeof num === "number")).toBe(true);
+  });
+
+  it("throws when dropping elements from an iterator that errors before the drop limit", () => {
```

Comment:
I'm not sure there's a concept like "before the drop the limit" here. Since `.drop()` removes elements from the beginning of the iterator, any iterator that eventually throws will still throw even after we `.drop()`.

This is still a worthwhile test though, and to make things interesting we could try different limits (using `it.each`). Additionally, we should highlight that the throw happens when we try to _consume_ the new iterator, not when we create it, so something like:

```js
// No throw at this stage:
const newIterator = iterator.drop(...);

// Throws when we traverse:
expect(() => [...newIterator]).toThrow();
```